src/components/related/ProductCardList.jsx

```diff
@@ -3,35 +3,40 @@ import axios from 'axios';
 import styles from './related.module.css';
 import { API_KEY } from '../../../config/config.js';
 import ProductCard from './ProductCard.jsx';
-
+import {useProductContext} from '../../context/ProductContext.jsx';
 
 const ProductCardList = () =>{
-  const [relatedID , setRelatedID] = useState([]);
-  const [overviewProduct, setOverviewProduct] = useState({});
+  const { related } = useProductContext();
+  // const [relatedID , setRelatedID] = useState([]);
+
+  // console.log(styles)
+
+  // const [overviewProduct, setOverviewProduct] = useState({});
 
-  const API_URL = 'https://app-hrsei-api.herokuapp.com/api/fec2/hr-rfp/products'
-  const options = {
-    headers: {'Authorization': API_KEY}
-  };
+  // const API_URL = 'https://app-hrsei-api.herokuapp.com/api/fec2/hr-rfp/products'
+  // const options = {
+  //   headers: {'Authorization': API_KEY}
+  // };
 
-  useEffect(() => {
-    // get the related products' id
-    axios.get(`${API_URL}/40365/related`,options)
-    .then(response => setRelatedID(response.data))
+  // useEffect(() => {
+  //   // get the related products' id
+  //   axios.get(`${API_URL}/40365/related`,options)
+  //   .then(response => setRelatedID(response.data))
 
-    // get the overview product info
-    axios.get(`${API_URL}/40365`,options)
-    .then(response => setOverviewProduct(response.data))
-  }, []);
+  //   // get the overview product info
+  //   axios.get(`${API_URL}/40365`,options)
+  //   .then(response => setOverviewProduct(response.data))
+  // }, []);
 
   // console.log(relatedID)
 
   return (
     <div className={styles.productCardList}>
-      {relatedID.map((result, index) => {
+      {Object.values(related).map((result, index) => {
+        // console.log(result)
```

Comment:
You have a lot of commented out code throughout these files and it makes it a little bit hard to follow. If you have a section of code to the point that it is working it would be nice to get rid of all console.logs. 

As far as the other commented out code goes I am kind of the same way and don't like getting rid of code in case I need it again in the future. I have been trying to force myself to not allow this to happen though because it makes it harder to follow. I would encourage you to look through your commented out code and remove any that you are pretty sure you won't need to use anymore.

Another option for code that you won't need to use again for the current project but you want to remember how you did it you could copy and paste to your engineering journal.

Reply:
The comment-out code is kept for possible future optimization. I think the way we grouping all the data together and then send to the client is not really efficient. I will move all the comment-out code to the very button.

Reply:
You are right there is definitely lots of room to improve the server code and optimize how we want to get the data we need.